### PR TITLE
feat(chatbar): add autocomplete to chatbar

### LINKED ISSUE #5151: Add @ mention functionality to chatbar

### PR DESCRIPTION
**Pull request by jasonwoodland**
### What this PR does 📖
- Add ChatbarAutocomplete component
- Username autocompletion in conversations triggered by typing '@'

### Which issue(s) this PR fixes 🔨
- Resolve #5151

### Special notes for reviewers 🗒️
We're going to add rich text insertion in a future PR! 🙇

### CODE REVIEW

**Review by josephmcg — CHANGES_REQUESTED** [review]

File: components/interactables/Editable/Editable.vue [hunk 1]
```diff
@@ -304,6 +305,7 @@ const Editable = Vue.extend({
       const messageBox = this.$refs?.editable as HTMLElement
       this.currentPosition = Cursor.getCurrentCursorPosition(messageBox)
       this.currentRange = getCurrentRange()
+      this.updateAutocomplete()
```
[the remark this answers is not in the document]
Comment (jasonwoodland) [reply components/interactables/Editable/Editable.vue]: Makes sense, I checked for presence of a `$listener`

**Comment by jasonwoodland** [comment]
@josephmcg made some improvements!

**Review by molimauro — APPROVED** [review]
Great!!

**Review by molimauro — APPROVED** [review]

**Review by stavares843 — COMMENTED** [review]
image


not sure why, sometimes the auto complete component layout appears like this 

instead of this

image

Chrome, macOS

**Review by stavares843 — COMMENTED** [review]
Captura de ecrã 2022-10-07, às 16 14 34

the satellite avatar doesn't seem in the right size

if you compare with other places, is not as small

**Review by stavares843 — COMMENTED** [review]
minor misalignment here

Captura de ecrã 2022-10-07, às 16 28 07

**Review by stavares843 — COMMENTED** [review]
there's no autocomplete when we edit a message is that for a future PR?

same question for when we send a @ name in the chat, is not bold or clickable to show the quick profile or similar for example

**Review by stavares843 — COMMENTED** [review]
Captura de ecrã 2022-10-07, às 16 29 48


do we need this "" before we start to search?

File: components/views/chat/chatbar/autocomplete/Autocomplete.less [hunk 2]
```diff
@@ -0,0 +1,34 @@
+.chatbar-autocomplete {
+  position: absolute;
+  bottom: 100%;
+  left: 0;
+  right: 0;
+  // background-color: @midground;
```
Comment (stavares843) [inline components/views/chat/chatbar/autocomplete/Autocomplete.less:6]: do we need this commented out? 🔨

**Comment by stavares843** [comment]
@molimauro what do you think? should we address this before or merge and address it after? is usable so, just some minor bugs

**Review by phillsatellite — APPROVED** [review]

**Comment by phillsatellite** [comment]
Besides bugs Sara and I found above, approved 👍

**Comment by molimauro** [comment]
> @molimauro what do you think? should we address this before or merge and address it after? is usable so, just some minor bugs

Seems that those are all minor problems/adjustments that we can address later. I think we can merge this!

Do you want me to add the tickets or could you do that? Thanks

**Comment by stavares843** [comment]
sounds good! will merge and add tickets 🔨

**Comment by stavares843** [comment]
added 🔨

**Pull request merged**